common/cluster/archival_status.go

```diff
@@ -47,38 +48,32 @@ const (
 )
 
 // NewArchivalConfig constructs a new valid ArchivalConfig
-func NewArchivalConfig(status ArchivalStatus, defaultBucket string, enableReadFromArchival bool) *ArchivalConfig {
+func NewArchivalConfig(
+	status ArchivalStatus,
+	enableReadFromArchival bool,
+	domainDefaultStatus ArchivalStatus,
+	domainDefaultURI string,
+) *ArchivalConfig {
 	ac := &ArchivalConfig{
 		status:                 status,
-		defaultBucket:          defaultBucket,
 		enableReadFromArchival: enableReadFromArchival,
+		domainDefaultStatus:    domainDefaultStatus,
+		domainDefaultURI:       domainDefaultURI,
 	}
-	if !ac.isValid() {
-		return &ArchivalConfig{
-			status:                 ArchivalDisabled,
-			defaultBucket:          "",
-			enableReadFromArchival: false,
-		}
+	if !ac.isDomainDefaultValid() {
+		ac.domainDefaultStatus = ArchivalDisabled
+		ac.domainDefaultURI = ""
 	}
 	return ac
 }
 
-// GetDefaultBucket returns the default bucket for ArchivalConfig
-func (a *ArchivalConfig) GetDefaultBucket() string {
-	return a.defaultBucket
-}
-
 // GetArchivalStatus returns the archival status for ArchivalConfig
 func (a *ArchivalConfig) GetArchivalStatus() ArchivalStatus {
 	return a.status
 }
 
 // ConfiguredForArchival returns true if cluster is configured to handle archival, false otherwise.
-// If cluster is configured for archival then defaultBucket will be set.
 func (a *ArchivalConfig) ConfiguredForArchival() bool {
-	if !a.isValid() {
-		return false
-	}
 	return a.status != ArchivalDisabled
```

Comment:
Is is a mistake on my part, but lets just treat ArchivalPaused as not configured either. Currently I want paused == disabled throughout the system.